packages/api/src/store/rabbit-queue.ts

```diff
@@ -0,0 +1,104 @@
+import amqp from "amqp-connection-manager";
+import { ChannelWrapper, AmqpConnectionManager } from "amqp-connection-manager";
+import { Channel, ConsumeMessage, Options } from "amqplib";
+
+const QUEUE_NAME = "webhook_default_queue";
+const EXCHANGE_NAME = "webhook_default_exchange";
+
+export default class MessageQueue {
+  private channel: ChannelWrapper;
+  private connection: AmqpConnectionManager;
+
+  constructor() {}
+
+  public async connect(url: string): Promise<void> {
+    // Create a new connection manager
+    this.connection = await amqp.connect([url]);
+    this.connection.on("connect", () => console.log("AMQP Connected!"));
+    this.connection.on("disconnect", (err: Error) =>
+      console.log("AMQP Disconnected.", err)
+    );
+    this.connection.on("error", (error: Error) => {
+      console.error(`Connection error: ${error}`);
+    });
+
+    this.channel = await this.connection.createChannel({
+      json: true,
+      setup: async function (channel: Channel) {
+        await Promise.all([
+          channel.assertQueue(QUEUE_NAME),
+          // channel.assertQueue("delayedQueue", {
+          //   messageTtl: 5000,
+          //   deadLetterExchange: EXCHANGE_NAME
+          // }),
+          channel.assertExchange(EXCHANGE_NAME, "topic", { durable: true }),
+          channel.prefetch(1),
+          channel.bindQueue(QUEUE_NAME, EXCHANGE_NAME, "#"),
+          // channel.consume(QUEUE_NAME, handleMessage)
+        ]);
+        // return channel.assertQueue('rxQueueName', {durable: true});
+      },
+    });
+  }
+
+  public async close(): Promise<void> {
+    await this.connection.removeAllListeners("connect");
+    await this.connection.removeAllListeners("disconnect");
+    await this.connection.removeAllListeners("error");
+    await this.connection.close();
+  }
+
+  public ack(data: any): void {
+    this.channel.ack(data);
+  }
+
+  public async consume(func: (msg: ConsumeMessage) => void): Promise<void> {
```

Comment:
**nit:** Just a trip, might not be worth it if it's more complex than it seems (not much experience with RabbitMQ): Would be nice to provide an `async` version of this which auto acks/nacks the msg based on the success or failure of the async function. 

Say, receive an 
```ts
asyncFunc: (msg: ConsumeMessage) => Promise<void>
``` 
here and send 
```ts
(msg) => asyncFunc(msg).then(() => channel.ack()).catch(() => channel.nack())
```
to the `channel.consume` function.